numpy/lib/tests/test_arraysetops.py

```diff
@@ -563,6 +563,50 @@ def test_unique_axis(self):
         result = np.array([[-0.0, 0.0]])
         assert_array_equal(unique(data, axis=0), result, msg)
 
+    def test_unique_axis_zeros(self):
+        # issue 15559
+        single_zero = np.empty(shape=(2, 0), dtype=np.int8)
+        uniq, idx, inv, cnt = unique(single_zero, axis=0, return_index=True,
+                                     return_inverse=True, return_counts=True)
+
+        # there's 1 element of shape (0,) along axis 0
+        assert_equal(uniq.dtype, single_zero.dtype)
+        msg = "Unique with shape=(2, 0) and axis=0 failed"
+        assert_array_equal(uniq, np.empty(shape=(1, 0)), msg)
```

Comment:
These messages don't seem to match the error messages.

Normally we don't bother with this type of message anyway, as its easy to infer from the stack trace.

Reply:
Nevermind, misread that. Still think that these messages are probably overkill, and make the test harder to follow

Reply:
Yeah, you can just skip the message. With pytest now, we get a huge amount of information already.

Reply:
Ah, ok 👍